app/views/admin/widget/cards/_form.html.erb

```diff
@@ -1,14 +1,58 @@
+<%= render "admin/shared/globalize_locales", resource: @card %>
+
 <%= form_for [:admin, @card] do |f| %>
+
+  <% @card.globalize_locales.each do |locale| %>
+    <%= hidden_field_tag "delete_translations[#{locale}]", 0 %>
+  <% end %>
+
   <div class="small-12 medium-6">
-    <%= f.text_field :label %>
+    <%= f.label :label %>
+    <% @card.globalize_locales.each do |locale| %>
+      <% globalize(locale) do %>
+        <%= f.text_field "label_#{locale}",
+                         class: "js-globalize-attribute",
+                         data: {locale: locale},
+                         style: display_translation?(locale),
+                         label: false %>
```

Comment:
A couple of suggestions regarding the text fields.

First, instead of using a generic label which isn't linked to a text field, what about removing the `f.label` lines and using something like:
```ruby
 <%= f.text_field "title_#{locale}",
                  placeholder: t("admin.banners.banner.title"),
                  class: "js-globalize-attribute",
                  data: {js_banner_title: "js_banner_title",
                         locale: locale},
                  style: display_translation?(locale),
                  label: t("admin.banners.banner.title"),
                  label_options: {
                    style: display_translation?(locale),
                    class: "js-globalize-attribute",
                    data: { locale: locale }
                   } %>
```
This way we can keep the benefits of using HTML labels (clicking on the label takes you to the field, ...).

Second, how about extracting this logic into a method? Maybe something like `f.translated_text field`?

Reply:
> This way we can keep the benefits of using HTML labels (clicking on the label takes you to the field, ...).

I wasn't aware of this behavior. Sure, I'll see if it works with the current JS implementation and change it.

> Second, how about extracting this logic into a method? Maybe something like f.translated_text field?

I thought about this a while ago, and started coding something. The tricky part is merging the relevant options. Here is what I was thinking:

```ruby
module TranslatableFormHelper
  def translatable_form_for(record, options = {}, &block)
    form_for(record, options.merge(builder: TranslatableFormBuilder), &block)
  end
end

class TranslatableFormBuilder < ActionView::Helpers::FormBuilder
  def translatable_text_field(method, _options = {})
    options = _options.dup
    options[:class] = (options[:class] || "") + " js-globalize-attribute" 
    options[:data] = (options[:data] || {}).merge(locale: locale)
    options[:label] = false
    options[:style] = display_translation?(locale)

    @template.capture do
      @object.globalize_locales.each do |locale|
        globalize(locale) do
          @template.concat do
            text_field(method, options.merge(title: _options[:title] + "_" + locale))
          end
        end
      end
    end

    text_field(method)
  end
```

Reply:
I think we're using a custom form builder (check `config/initializers/foundation_rails_helper.rb`). Maybe we could add it there, next to `cktext_area`?

I'd keep the `each` loop and the `capture` and `globalize` part in another method (maybe even outside the builder? I'm not sure), but since it's a matter of personal preference, I'll leave it up to you.

Reply:
Well another reason to use a custom `FormBuilder` is to handle the "deleted_<locale>" tags that every translatable form needs.

I'll consider your other suggestion, code something up quickly and see what you think.

Reply:
I think you're right! Just one thing; instead of:

`class TranslatableFormBuilder < ActionView::Helpers::FormBuilder`

I think it would be safer to inherit from:

`class TranslatableFormBuilder < FoundationHelpers::FormBuilder`

Since that's the builder we use in the rest of the application (or so I think).

Reply:
Here is a working implementation of the FormBuilder solution: https://github.com/AyuntamientoMadrid/consul/commit/7e38dc7c0caf5d8daa29c43aa8c96790c421b58b. What do you think @javierm? 

I'll have to open a new PR to apply these changes to all the translatable forms in admin. @javierm do you want to merge this branch as it is, and wait for the dedicated refactor PR to make these changes? Or do you want me to add the changes here (and then also open a refactor PR to make the changes to other forms)?

Reply:
@mlovic Looks great! :tada:. One small thing: would it be possible to remove the `f.label` calls and include something like what's suggested in https://github.com/AyuntamientoMadrid/consul/pull/1588#discussion_r208894203 in the `translatable_field` method? I tested the `label_options` locally and they seemed to work fine with globalize, but maybe I missed something.

Regarding PRs, I'll leave a comment in #1602.

Reply:
Regarding where to add the translatable form builder, IMHO you could either add it to #1602 or open a new one which adds the builder and applies it to the already merged banners. Up to you!

Reply:
Several of these suggestions have been implemented in https://github.com/AyuntamientoMadrid/consul/pull/1613.